Use `torch.empty()` instead of `torch.tensor()` in `torch.nn.Parameter`

cbalioglu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66486

The newly-introduced Python dispatcher mode (`__torch_dispatch__`) does not have support for `torch.tensor()` (see #64360) and this causes friction in the user experience if some `nn.Modules` use `torch.tensor()` either implicitly or explicitly.

This PR replaces calls to `torch.tensor()` in `Parameter`, `UninitializedParameter`, and `UninitializedBuffer` with an equivalent call to `torch.empty()` which serves the same purpose and is syntactically more readable.

Differential Revision: [D31575587](https://our.internmc.facebook.com/intern/diff/D31575587/)

cbalioglu committed:
Use `torch.empty()` instead of `torch.tensor()` in `torch.nn.Parameter`

The newly-introduced Python dispatcher mode (`__torch_dispatch__`) does not have support for `torch.tensor()` (see #64360) and this causes friction in the user experience if some `nn.Modules` use `torch.tensor()` either implicitly or explicitly.

This PR replaces calls to `torch.tensor()` in `Parameter`, `UninitializedParameter`, and `UninitializedBuffer` with an equivalent call to `torch.empty()` which serves the same purpose and is syntactically more readable.

Differential Revision: [D31575587](https://our.internmc.facebook.com/intern/diff/D31575587/)

[ghstack-poisoned]

cbalioglu committed:
Update on "Use `torch.empty()` instead of `torch.tensor()` in `torch.nn.Parameter`"


The newly-introduced Python dispatcher mode (`__torch_dispatch__`) does not have support for `torch.tensor()` (see #64360) and this causes friction in the user experience if some `nn.Modules` use `torch.tensor()` either implicitly or explicitly.

This PR replaces calls to `torch.tensor()` in `Parameter`, `UninitializedParameter`, and `UninitializedBuffer` with an equivalent call to `torch.empty()` which serves the same purpose and is syntactically more readable.

Differential Revision: [D31575587](https://our.internmc.facebook.com/intern/diff/D31575587/)

[ghstack-poisoned]

albanD reviewed (approved):
Sounds like an ok workaround until `torch.tensor()` is fixed.

cbalioglu committed:
Update on "Use `torch.empty()` instead of `torch.tensor()` in `torch.nn.Parameter`"


The newly-introduced Python dispatcher mode (`__torch_dispatch__`) does not have support for `torch.tensor()` (see #64360) and this causes friction in the user experience if some `nn.Modules` use `torch.tensor()` either implicitly or explicitly.

This PR replaces calls to `torch.tensor()` in `Parameter`, `UninitializedParameter`, and `UninitializedBuffer` with an equivalent call to `torch.empty()` which serves the same purpose and is syntactically more readable.

Differential Revision: [D31575587](https://our.internmc.facebook.com/intern/diff/D31575587/)

[ghstack-poisoned]
